Core/SubSystem.cpp

```diff
@@ -71,86 +74,100 @@ boost::property_tree::ptree SubSystem::save() const {
   return pt;
 }
 
-SubSystem::SubSystem(std::vector<int> recoilS,
-                     std::vector<std::vector<int>> finalStates)
-    : _recoilState(recoilS), _finalStates(finalStates) {
-
+SubSystem::SubSystem(
+    const std::vector<unsigned int>& recoilS,
+    const std::vector<std::vector<unsigned int>>& finalStates) {
+  SetFinalStates(finalStates);
+  SetRecoilState(recoilS);
   _title = to_string();
   // LOG(trace) << "SubSystem::SubSystem() | Creating sub system "<<title;
 }
 
-/// Constructor for an isobar model in which it is assumed that particles
+/*/// Constructor for an isobar model in which it is assumed that particles
 /// always decay to two final states.
-SubSystem::SubSystem(std::vector<int> recoilS, std::vector<int> finalA,
-                     std::vector<int> finalB)
+SubSystem::SubSystem(const std::vector<unsigned int>& recoilS, const
+std::vector<unsigned int>& finalA,
+                     const std::vector<unsigned int>& finalB)
     : _recoilState(recoilS) {
-  std::vector<std::vector<int>> tmp;
+  std::vector<std::vector<unsigned int>> tmp;
   tmp.push_back(finalA);
   tmp.push_back(finalB);
   SetFinalStates(tmp);
 
   _title = to_string();
   // LOG(trace) << "SubSystem::SubSystem() | Creating sub system "<<title;
-}
+}*/
 
 std::string SubSystem::to_string() const {
   std::stringstream stream;
 
-  for (auto j = _finalStates.begin(); j != _finalStates.end(); ++j) {
-    for (auto i : *j)
-      stream << std::to_string(i);
-    if (j != _finalStates.end() - 1)
-      stream << "_";
+  for (auto const& j : _finalStates) {
+    for (auto const& i : j) stream << std::to_string(i);
+    stream << "_";
   }
-  stream << "_vs_";
-  for (auto i : _recoilState)
-    stream << std::to_string(i);
+  stream << "vs_";
+  for (auto i : _recoilState) stream << std::to_string(i);
```

Comment:
Some parts of the diff is only due to a line break after control statements. I didn't explicitly find it in the LLVM standard but all the examples have a line break.